src/example/Simplest.js

```diff
@@ -0,0 +1,71 @@
+import ActionManager from "../action/ActionManager.js";
+import DoubleJumpFilter from "./filters/DoubleJumpFilter.js";
+
+function main() {
+  // App specific setup of the ActionManager
+
+  let actionManager = new ActionManager();
+
+  // Load a custom filter
+  actionManager.loadFilter(DoubleJumpFilter, "/filter/double-jump");
```

Comment:
I think this type of input filtering needs to happen in application logic, not plugged into the input library. The double jump filter will presumably allow only jumps and double-jumps and then throw out all the other attempts to jump. However, knowing when a jump is valid or when a double-jump is valid requires knowledge about the application state, which filters should not rely on. For example, a double-jump might need to be pressed within a certain interval of a jump, but if the character landed on a platform in the meantime, then their second button press is not really a double-jump - it's just a jump. Similarly, if the character jumps off a very high platform, the input library would need to know when the character is grounded again before allowing any more valid jumps. Since filters are configurable in a binding definition (and can be added or removed), it's best to leave them as things like reversing axes, low-pass filters, configuring when a trigger will return true for a boolean action (e.g. simple threshold vs dynamic threshold). In short I don't think application logic should load or apply filters.

Reply:
I agree with this

Reply:
I agree that most filters will be lower level than "doublejump" and that those filters should be written once and included in the action-input lib.
There will be app-specific filters that aren't generic that the web app needs to supply.
For example, I (acting as a web dev) could write a new filter that watches axis input from thumbpads (like on a Vive controller), recognizes certain types of swipes, and emit high level actions. So, in my web app if you swirl your thumb counter-clockwise it could emit a `deploy-antelope` action but if you swirl your thumb clockwise it could mean 'deploy-chickens'.
So, my goal is to allow web devs to write filters, specify them them in their maps, and receive actions using the same API as the included filters.